force-app/main/default/classes/ACCT_IndividualAccounts_TEST.cls

```diff
@@ -445,7 +445,198 @@ private class ACCT_IndividualAccounts_TEST {
 
         Account assertAccount = [SELECT Id, RecordType.Name, Name FROM Account LIMIT 1];
         System.assertEquals('Household Account', assertAccount.RecordType.Name);
-        System.assertEquals(con.FirstName + ' ' + con.LastName + ' Household', assertAccount.Name);
+        System.assertEquals(con.FirstName + ' ' + con.LastName + ' ' + Label.DefaultHouseholdName, assertAccount.Name);
+    }
+
+    /*********************************************************************************************************
+     * @description Insert a new contact and the default account record type is Household
+     * Make sure the Account's name is set based on account naming setting for a user in Spanish.
+     * Test Scenario: T-3904056
+     */
+    @isTest
+    private static void newContactNewHHAccSpanish() {
+        System.runAs(ACCT_IndividualAccounts_TEST.returnUserInSpanish()) {
+            Hierarchy_Settings__c edaSettings = new Hierarchy_Settings__c(
+                Account_Processor__c = UTIL_Describe.getHhAccRecTypeID(),
+                Household_Account_Naming_Format__c = Label.acctNamingOther,
+                Automatic_Household_Naming__c = true,
+                Household_Other_Name_Setting__c = Label.lastNameHH
+            );
+
+            UTIL_CustomSettingsFacade.getSettingsForTests(edaSettings);
+
+            Contact con = UTIL_UnitTestData_API.getContact();
+            con.Email = 's@salesfor.com';
+            con.Preferred_Email__c = 'Correo electrónico laboral';
+
+            Test.startTest();
+            insert con;
+            Test.stopTest();
+
+            Account assertAccount = [SELECT Id, RecordType.Name, Name FROM Account LIMIT 1];
+            System.assertEquals('Household Account', assertAccount.RecordType.Name);
+            System.assertEquals('Domicilio de ' + con.LastName, assertAccount.Name);
```

Comment:
Hi @ZeeshanSyedH , I'm just getting started reviewing some of the changes, but I noticed this assertion passes, however I'm still wondering how this worked. I repeated the steps in my 2gp org with a user with Spanish language. The account name should be con.Lastname + 'Cuenta'.
This is because the Household_Other_Name_Setting__c is set to Label.lastNameHH (<value>{!LastName} Household</value>). Please take a look at the screenshot:
![image](https://user-images.githubusercontent.com/63684581/152409551-1b468069-fa96-4f36-ae99-adec55b20e3e.png)

Reply:
Within the test context here, since we consume the label as the running user(language: spanish) - the Household_Other_Name_Setting__c field was set too "Domicilio de {!LastName}" ([reference](https://github.com/SalesforceFoundation/EDA/blob/main/force-app/main/default/translations/es.translation-meta.xml#L252))